CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+There are many ways to contribute to Highlight.js. Reporting issues is super useful. Feel free to work on an open issue. You can also help by writing [a plugin](https://highlightjs.readthedocs.io/en/latest/plugin-api.html), adding [a 3rd party grammar](https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md), or creating [a theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html).
+
+If you need support, you can [open a GitHub issue](https://github.com/highlightjs/highlight.js/issues/new/choose).
+
+
+## Philosophy
+
+Highlight.js makes a strong effort to keep the core engine, it's usage, and it's feature set clean and simple.  We aim to support the happy path (the 90%), not the often innumerable edge cases (the last 10%).  When you're on the happy path everything "just works" with a single line of code.  We do our best to allow those familiar with JavaScript to easily handle the remaining 10% via small extensions or plug-ins.
+
+Some bullet points:
+
+- Our [lack of line numbers is a feature](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). It's usually pretty simple to add them yourself ([example](https://github.com/taufik-nurrohman/highlight.ln.js)).
+- We [embrace plug-ins and extensions](https://github.com/highlightjs/highlight.js/issues/2225) rather than adding lots of tiny features and configuration options.
+- We are much more than a keyword highlighter. We do make attempts to understand context...
+- ...Yet we are not a full parser for any language, we're still just a fancy nested pattern matcher.
+- Our [auto-detect is sadly not powered by magic unicorn dust](https://github.com/highlightjs/highlight.js/issues/1213). It's best effort, not best in class.
+
+
+## Feature Requests
+Feature requests are always welcome but many times the answer is:
+
+> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.
+
+If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
+
+
+## Reporting an Issue
+(Misc.)
+- If you found a language highlighting issue, use [this JSFiddle](https://jsfiddle.net/ajoshguy/2bmdswn6/) to create a test case.
+
+
+## Fixing an Issue (PRs)
+If you're new to contributing to open-source, have a look at [this GitHub Guide](https://guides.github.com/activities/forking). It explains the general process of GitHub collaboration.
+
+If you feel comfortable with our [prerequisites](#prerequisites), you can grab any issue [marked `beginner friendly`](https://github.com/highlightjs/highlight.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22).
```

Comment:
The label changed to "good first issue".

Reply:
```suggestion
If you feel comfortable with our [prerequisites](#prerequisites), you can grab any issue [marked "good first issue"](https://github.com/highlightjs/highlight.js/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
```